File: hotel_room/hotel_room.py
# https://programmers.co.kr/learn/courses/30/lessons/64063

import sys
sys.setrecursionlimit(10**6) 

def find_next_room(m, room):
    if room not in m:
        m[room] = room + 1
        return room
    m[room] = find_next_room(m, m[room])
    return m[room]

def solution(k, room_number):
    answer = []
    m = dict()
    for room in room_number:
        next_room = find_next_room(m, room)
        answer.append(next_room)
    return answer

File: hotel_room/hotel_room.cpp
#include <string>
#include <vector>
#include <iostream>
#include <map>

using namespace std;


map<long long, long long>m; 

long long find_next(long long x) {
	if (m[x] == 0) {
		m[x] = x + 1;
		return x;
	}
	m[x] = find_next(m[x]);
	return m[x];
}

vector<long long> solution(long long k, vector<long long> room_number) {
    vector<long long> answer;

	for (auto room : room_number) {
		long long n = find_next(room);
		answer.push_back(n);
	}

    return answer;
}

int main() {
	vector<long long> v {1, 3, 4, 1, 3, 1};
	vector<long long> answer = solution(10, v);
	for (auto i : answer) {
		cout << i << "\n"
	}
}